parquet-column/src/main/java/org/apache/parquet/column/page/DataPage.java

```diff
@@ -37,6 +45,33 @@ public int getValueCount() {
     return valueCount;
   }
 
+  /**
+   * @return the index of the first row in this page
+   * @throws NotInPageFilteringModeException
+   *           if page filtering mode is not active
+   * @see PageReadStore#isInPageFilteringMode()
+   */
+  public long getFirstRowIndex() {
+    if (firstRowIndex < 0) {
+      throw new NotInPageFilteringModeException("First row index is not available");
```

Comment:
Why does this throw an exception for the filtering mode? The reader mode doesn't matter to this. It either has a starting index and row count or it doesn't.

I think a better API would be to avoid throwing an exception by using Optional, Long/null, or -1 to signal that the page can't report this. Throwing an exception can cause a runtime error when an expectation isn't met, while returning an Option forces callers to handle the case where the option is empty.